src/containers/Tenant/QueryEditor/QueryExplain/QueryExplain.js

```diff
@@ -116,7 +116,7 @@ function QueryExplain(props) {
         if (!props.ast && activeOption === ExplainOptionIds.ast) {
             props.astQuery();
         }
-    }, [activeOption]);
+    }, [activeOption, props.ast]);
```

Comment:
I'd probably move `renderAstExplain` to the separate component.
And this `useEffect` seems to be redundant. You may simply check if-statement in main flow.

Reply:
`QueryExplain` should be split to several components and migrated to TS, but it isn't the issue of this PR.

As for the `useEffect`, I don't agree, without `useEffect` request will be sent at least twice as request toggles component loading state which makes component to rerender. Also it could be bugged in case of unsuccessful requests. Request fail -> component rerender because of a new error -> request is sent again -> loading state toggled -> new request is sent, infinite loader is shown instead of an error

Reply:
We may get rid of `useEffect` when move Ast to new component: just `error` should be added into if-statement.
Loader is in parent component, so when it is shown, `<Ast/>` would not be rendered. When loading is done, `props.ast` or `props.error` is here.

Reply:
I tried separating `AST`. Without hooks it still causes infinite data fetches. I don't know how to ensure single fetch without wrapping either fetch or component itself in a hook. I'd prefer fetch data in hook and separate `AST` with complex `QueryExplain` component refactoring

Reply:
Ok, let's think about it when refactor `QueryExplain`.